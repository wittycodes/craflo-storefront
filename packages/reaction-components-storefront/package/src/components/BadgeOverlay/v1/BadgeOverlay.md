### Overview
Renders an overlay that displays a primary and secondary badge for a product, if appropriate.

#### Usage

##### Sold out
When sold out, the overlay and its children are faded to 50% visibility.

```jsx
<BadgeOverlay
  product={{
    isSoldOut: true,
    isBackorder: false,
    isOnSale: false,
    isLowQuantity: false,
    isBestseller: false
  }}
>
  <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library logo" />
</BadgeOverlay>
```

##### Backorder

```jsx
<BadgeOverlay
  product={{
    isSoldOut: true,
    isBackorder: true,
    isOnSale: false,
    isLowQuantity: false,
    isBestseller: false
  }}
>
  <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library logo" />
</BadgeOverlay>
```

##### On sale

```jsx
<BadgeOverlay
  product={{
    isSoldOut: false,
    isBackorder: false,
    isOnSale: true,
    isLowQuantity: false,
    isBestseller: false
  }}
>
  <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library" />
</BadgeOverlay>
```

##### On sale with low quantity secondary badge

```jsx
<BadgeOverlay
  product={{
    isSoldOut: false,
    isBackorder: false,
    isOnSale: true,
    isLowQuantity: true,
    isBestseller: false
  }}
>
  <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library" />
</BadgeOverlay>
```

##### On sale with best seller secondary badge

```jsx
<BadgeOverlay
  product={{
    isSoldOut: false,
    isBackorder: false,
    isOnSale: true,
    isLowQuantity: false,
    isBestseller: true
  }}
>
  <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library" />
</BadgeOverlay>
```

##### Low quantity

```jsx
<BadgeOverlay
  product={{
    isSoldOut: false,
    isBackorder: false,
    isOnSale: false,
    isLowQuantity: true,
    isBestseller: false
  }}
>
  <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library" />
</BadgeOverlay>
```

##### Low quantity with best seller secondary badge

```jsx
<BadgeOverlay
  product={{
    isSoldOut: false,
    isBackorder: false,
    isOnSale: false,
    isLowQuantity: true,
    isBestseller: true
  }}
>
  <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library" />
</BadgeOverlay>
```

##### Best seller

```jsx
<BadgeOverlay
  product={{
    isSoldOut: false,
    isBackorder: false,
    isOnSale: false,
    isLowQuantity: false,
    isBestseller: true
  }}
>
  <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library" />
</BadgeOverlay>
```

### Theme

Assume that any theme prop that does not begin with "rui" is within `rui_components`. See [Theming Components](./#!/Theming%20Components).

| Theme Prop                                           | Default  | Description                                                                   |
|------------------------------------------------------|----------|-------------------------------------------------------------------------------|
| BadgeOverlay.fadedOpacity                            | 0.5      | Opacity of the entire component and its contents when the status is "soldOut" |
| BadgeOverlayPrimaryBadge.backgroundColor_backorder   | coolGrey | Label background color for the primary badge when the status is "backorder"   |
| BadgeOverlayPrimaryBadge.backgroundColor_bestseller  | teal     | Label background color for the primary badge when the status is "bestseller"  |
| BadgeOverlayPrimaryBadge.backgroundColor_lowQuantity | coolGrey | Label background color for the primary badge when the status is "lowQuantity" |
| BadgeOverlayPrimaryBadge.backgroundColor_sale        | red300   | Label background color for the primary badge when the status is "sale"        |
| BadgeOverlayPrimaryBadge.backgroundColor_soldOut     | coolGrey | Label background color for the primary badge when the status is "soldOut"     |
| BadgeOverlayPrimaryBadge.borderBottomLeftRadius      | 4px      | Bottom left corner border radius for the primary badge                        |
| BadgeOverlayPrimaryBadge.borderBottomRightRadius     | 4px      | Bottom right corner border radius for the primary badge                       |
| BadgeOverlayPrimaryBadge.borderTopLeftRadius         | 4px      | Top left corner border radius for the primary badge                           |
| BadgeOverlayPrimaryBadge.borderTopRightRadius        | 4px      | Top right corner border radius for the primary badge                          |
| BadgeOverlayPrimaryBadge.color_backorder             | white    | Label text color for the primary badge when the status is "backorder"         |
| BadgeOverlayPrimaryBadge.color_bestseller            | white    | Label text color for the primary badge when the status is "bestseller"        |
| BadgeOverlayPrimaryBadge.color_lowQuantity           | white    | Label text color for the primary badge when the status is "lowQuantity"       |
| BadgeOverlayPrimaryBadge.color_sale                  | white    | Label text color for the primary badge when the status is "sale"              |
| BadgeOverlayPrimaryBadge.color_soldOut               | white    | Label text color for the primary badge when the status is "soldOut"           |
| BadgeOverlayPrimaryBadge.offsetBottom                | auto     | The primary badge is absolutely positioned with this bottom offset            |
| BadgeOverlayPrimaryBadge.offsetLeft                  | 8px      | The primary badge is absolutely positioned with this left offset              |
| BadgeOverlayPrimaryBadge.offsetRight                 | auto     | The primary badge is absolutely positioned with this right offset             |
| BadgeOverlayPrimaryBadge.offsetTop                   | 8px      | The primary badge is absolutely positioned with this top offset               |
| BadgeOverlayPrimaryBadge.paddingBottom               | 4px      | Bottom padding for the primary badge                                          |
| BadgeOverlayPrimaryBadge.paddingLeft                 | 8px      | Left padding for the primary badge                                            |
| BadgeOverlayPrimaryBadge.paddingRight                | 8px      | Right padding for the primary badge                                           |
| BadgeOverlayPrimaryBadge.paddingTop                  | 4px      | Top padding for the primary badge                                             |
| BadgeOverlaySecondaryBadge.borderBottomLeftRadius    | 4px      | Bottom left corner border radius for the secondary badge                      |
| BadgeOverlaySecondaryBadge.borderBottomRightRadius   | 4px      | Bottom right corner border radius for the secondary badge                     |
| BadgeOverlaySecondaryBadge.borderTopLeftRadius       | 4px      | Top left corner border radius for the secondary badge                         |
| BadgeOverlaySecondaryBadge.borderTopRightRadius      | 4px      | Top right corner border radius for the secondary badge                        |
| BadgeOverlaySecondaryBadge.color                     | coolGrey | Label text color for the secondary badge                                      |
| BadgeOverlaySecondaryBadge.offsetBottom              | auto     | The secondary badge is absolutely positioned with this bottom offset          |
| BadgeOverlaySecondaryBadge.offsetLeft                | auto     | The secondary badge is absolutely positioned with this left offset            |
| BadgeOverlaySecondaryBadge.offsetRight               | 8px      | The secondary badge is absolutely positioned with this right offset           |
| BadgeOverlaySecondaryBadge.offsetTop                 | 8px      | The secondary badge is absolutely positioned with this top offset             |
| BadgeOverlaySecondaryBadge.paddingBottom             | 4px      | Bottom padding for the secondary badge                                        |
| BadgeOverlaySecondaryBadge.paddingLeft               | 8px      | Left padding for the secondary badge                                          |
| BadgeOverlaySecondaryBadge.paddingRight              | 8px      | Right padding for the secondary badge                                         |
| BadgeOverlaySecondaryBadge.paddingTop                | 4px      | Top padding for the secondary badge                                           |
