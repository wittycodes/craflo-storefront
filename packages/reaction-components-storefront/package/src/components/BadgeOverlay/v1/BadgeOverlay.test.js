import React from "react";
import renderer from "react-test-renderer";
import BadgeOverlay from "./BadgeOverlay";

test("BadgeOverlay with sold out primary badge", () => {
  const component = renderer.create((
    <BadgeOverlay
      product={{
        isSoldOut: true,
        isBackorder: false,
        isOnSale: false,
        isLowQuantity: false,
        isBestseller: false
      }}
    >
      <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library logo" />
    </BadgeOverlay>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("BadgeOverlay with backorder primary badge", () => {
  const component = renderer.create((
    <BadgeOverlay
      product={{
        isSoldOut: true,
        isBackorder: true,
        isOnSale: false,
        isLowQuantity: false,
        isBestseller: false
      }}
    >
      <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library logo" />
    </BadgeOverlay>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("BadgeOverlay with on sale primary badge", () => {
  const component = renderer.create((
    <BadgeOverlay
      product={{
        isSoldOut: false,
        isBackorder: false,
        isOnSale: true,
        isLowQuantity: false,
        isBestseller: false
      }}
    >
      <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library logo" />
    </BadgeOverlay>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("BadgeOverlay with on sale primary badge and low quantity secondary badge", () => {
  const component = renderer.create((
    <BadgeOverlay
      product={{
        isSoldOut: false,
        isBackorder: false,
        isOnSale: true,
        isLowQuantity: true,
        isBestseller: false
      }}
    >
      <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library logo" />
    </BadgeOverlay>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("BadgeOverlay with on sale primary badge and best seller secondary badge", () => {
  const component = renderer.create((
    <BadgeOverlay
      product={{
        isSoldOut: false,
        isBackorder: false,
        isOnSale: true,
        isLowQuantity: false,
        isBestseller: true
      }}
    >
      <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library logo" />
    </BadgeOverlay>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("BadgeOverlay with low quantity primary badge", () => {
  const component = renderer.create((
    <BadgeOverlay
      product={{
        isSoldOut: false,
        isBackorder: false,
        isOnSale: false,
        isLowQuantity: true,
        isBestseller: false
      }}
    >
      <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library logo" />
    </BadgeOverlay>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("BadgeOverlay with low quantity primary badge and best seller secondary badge", () => {
  const component = renderer.create((
    <BadgeOverlay
      product={{
        isSoldOut: false,
        isBackorder: false,
        isOnSale: false,
        isLowQuantity: true,
        isBestseller: true
      }}
    >
      <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library logo" />
    </BadgeOverlay>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("BadgeOverlay with best seller primary badge", () => {
  const component = renderer.create((
    <BadgeOverlay
      product={{
        isSoldOut: false,
        isBackorder: false,
        isOnSale: false,
        isLowQuantity: false,
        isBestseller: true
      }}
    >
      <img src="/storefront-component-library-logo.svg" width="200" height="200" alt="Reaction Storefront Component Library logo" />
    </BadgeOverlay>
  ));

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
