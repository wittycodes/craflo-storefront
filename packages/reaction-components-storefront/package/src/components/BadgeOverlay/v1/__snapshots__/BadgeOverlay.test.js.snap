// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeOverlay with backorder primary badge 1`] = `
.c1 {
  font-size: 11px;
  height: auto;
  line-height: 16px;
  position: absolute;
  z-index: 1200;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  bottom: auto;
  left: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  right: auto;
  top: 8px;
  background-color: #5e7480;
  color: #ffffff;
}

.c0 {
  position: relative;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  color: inherit;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 11px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.25;
  padding: 0;
  position: relative;
  white-space: nowrap;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="BACKORDER"
  >
    <span
      className="c2"
    >
      Backorder
    </span>
  </div>
  <img
    alt="Reaction Storefront Component Library logo"
    height="200"
    src="/storefront-component-library-logo.svg"
    width="200"
  />
</div>
`;

exports[`BadgeOverlay with best seller primary badge 1`] = `
.c1 {
  font-size: 11px;
  height: auto;
  line-height: 16px;
  position: absolute;
  z-index: 1200;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  bottom: auto;
  left: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  right: auto;
  top: 8px;
  background-color: #8ce0c9;
  color: #ffffff;
}

.c0 {
  position: relative;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  color: inherit;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 11px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.25;
  padding: 0;
  position: relative;
  white-space: nowrap;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="BESTSELLER"
  >
    <span
      className="c2"
    >
      Best Seller
    </span>
  </div>
  <img
    alt="Reaction Storefront Component Library logo"
    height="200"
    src="/storefront-component-library-logo.svg"
    width="200"
  />
</div>
`;

exports[`BadgeOverlay with low quantity primary badge 1`] = `
.c1 {
  font-size: 11px;
  height: auto;
  line-height: 16px;
  position: absolute;
  z-index: 1200;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  bottom: auto;
  left: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  right: auto;
  top: 8px;
  background-color: #5e7480;
  color: #ffffff;
}

.c0 {
  position: relative;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  color: inherit;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 11px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.25;
  padding: 0;
  position: relative;
  white-space: nowrap;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="LOW_QUANTITY"
  >
    <span
      className="c2"
    >
      Low Inventory
    </span>
  </div>
  <img
    alt="Reaction Storefront Component Library logo"
    height="200"
    src="/storefront-component-library-logo.svg"
    width="200"
  />
</div>
`;

exports[`BadgeOverlay with low quantity primary badge and best seller secondary badge 1`] = `
.c1 {
  font-size: 11px;
  height: auto;
  line-height: 16px;
  position: absolute;
  z-index: 1200;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  bottom: auto;
  left: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  right: auto;
  top: 8px;
  background-color: #5e7480;
  color: #ffffff;
}

.c3 {
  font-size: 11px;
  height: auto;
  line-height: 16px;
  position: absolute;
  z-index: 1200;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  bottom: auto;
  color: #5e7480;
  left: auto;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  right: 8px;
  top: 8px;
}

.c0 {
  position: relative;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  color: inherit;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 11px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.25;
  padding: 0;
  position: relative;
  white-space: nowrap;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="LOW_QUANTITY"
  >
    <span
      className="c2"
    >
      Low Inventory
    </span>
  </div>
  <div
    className="c3"
  >
    <span
      className="c2"
    >
      Best Seller
    </span>
  </div>
  <img
    alt="Reaction Storefront Component Library logo"
    height="200"
    src="/storefront-component-library-logo.svg"
    width="200"
  />
</div>
`;

exports[`BadgeOverlay with on sale primary badge 1`] = `
.c1 {
  font-size: 11px;
  height: auto;
  line-height: 16px;
  position: absolute;
  z-index: 1200;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  bottom: auto;
  left: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  right: auto;
  top: 8px;
  background-color: #e54f5d;
  color: #ffffff;
}

.c0 {
  position: relative;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  color: inherit;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 11px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.25;
  padding: 0;
  position: relative;
  white-space: nowrap;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="SALE"
  >
    <span
      className="c2"
    >
      Sale
    </span>
  </div>
  <img
    alt="Reaction Storefront Component Library logo"
    height="200"
    src="/storefront-component-library-logo.svg"
    width="200"
  />
</div>
`;

exports[`BadgeOverlay with on sale primary badge and best seller secondary badge 1`] = `
.c1 {
  font-size: 11px;
  height: auto;
  line-height: 16px;
  position: absolute;
  z-index: 1200;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  bottom: auto;
  left: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  right: auto;
  top: 8px;
  background-color: #e54f5d;
  color: #ffffff;
}

.c3 {
  font-size: 11px;
  height: auto;
  line-height: 16px;
  position: absolute;
  z-index: 1200;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  bottom: auto;
  color: #5e7480;
  left: auto;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  right: 8px;
  top: 8px;
}

.c0 {
  position: relative;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  color: inherit;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 11px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.25;
  padding: 0;
  position: relative;
  white-space: nowrap;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="SALE"
  >
    <span
      className="c2"
    >
      Sale
    </span>
  </div>
  <div
    className="c3"
  >
    <span
      className="c2"
    >
      Best Seller
    </span>
  </div>
  <img
    alt="Reaction Storefront Component Library logo"
    height="200"
    src="/storefront-component-library-logo.svg"
    width="200"
  />
</div>
`;

exports[`BadgeOverlay with on sale primary badge and low quantity secondary badge 1`] = `
.c1 {
  font-size: 11px;
  height: auto;
  line-height: 16px;
  position: absolute;
  z-index: 1200;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  bottom: auto;
  left: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  right: auto;
  top: 8px;
  background-color: #e54f5d;
  color: #ffffff;
}

.c3 {
  font-size: 11px;
  height: auto;
  line-height: 16px;
  position: absolute;
  z-index: 1200;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  bottom: auto;
  color: #5e7480;
  left: auto;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  right: 8px;
  top: 8px;
}

.c0 {
  position: relative;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  color: inherit;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 11px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.25;
  padding: 0;
  position: relative;
  white-space: nowrap;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="SALE"
  >
    <span
      className="c2"
    >
      Sale
    </span>
  </div>
  <div
    className="c3"
  >
    <span
      className="c2"
    >
      Low Inventory
    </span>
  </div>
  <img
    alt="Reaction Storefront Component Library logo"
    height="200"
    src="/storefront-component-library-logo.svg"
    width="200"
  />
</div>
`;

exports[`BadgeOverlay with sold out primary badge 1`] = `
.c1 {
  font-size: 11px;
  height: auto;
  line-height: 16px;
  position: absolute;
  z-index: 1200;
  border-bottom-left-radius: 4px;
  border-bottom-right-radius: 4px;
  border-top-left-radius: 4px;
  border-top-right-radius: 4px;
  bottom: auto;
  left: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  right: auto;
  top: 8px;
  background-color: #5e7480;
  color: #ffffff;
}

.c0 {
  position: relative;
  opacity: 0.5;
}

.c2 {
  -webkit-font-smoothing: antialiased;
  color: inherit;
  font-family: 'Source Sans Pro','Helvetica Neue',Helvetica,sans-serif;
  font-size: 11px;
  font-style: normal;
  font-stretch: normal;
  font-weight: 700;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  line-height: 1.25;
  padding: 0;
  position: relative;
  white-space: nowrap;
}

<div
  className="c0"
>
  <div
    className="c1"
    type="SOLD_OUT"
  >
    <span
      className="c2"
    >
      Sold Out
    </span>
  </div>
  <img
    alt="Reaction Storefront Component Library logo"
    height="200"
    src="/storefront-component-library-logo.svg"
    width="200"
  />
</div>
`;
